pallets/subsocial-proxy/src/lib.rs

```diff
@@ -0,0 +1,103 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+pub use pallet::*;
+
+pub use crate::weights::WeightInfo;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+
+pub mod weights;
+
+#[frame_support::pallet]
+pub mod pallet {
+    use frame_support::{dispatch::RawOrigin, pallet_prelude::*, traits::Currency};
+    use frame_system::pallet_prelude::*;
+    use sp_runtime::traits::Zero;
+    use crate::weights::WeightInfo;
+
+    type BalanceOf<T> = <<T as pallet_proxy::Config>::Currency as Currency<
+        <T as frame_system::Config>::AccountId,
+    >>::Balance;
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config + pallet_proxy::Config {
+        type ProxyDepositBase: Get<BalanceOf<Self>>;
+
+        type ProxyDepositFactor: Get<BalanceOf<Self>>;
+
+        /// Weight information for extrinsics in this pallet.
+        type WeightInfo: WeightInfo;
+    }
+
+    #[pallet::pallet]
+    #[pallet::generate_store(pub(super) trait Store)]
+    pub struct Pallet<T>(_);
+
+    #[pallet::error]
+    pub enum Error<T> {
+        OnlyFirstProxyCanBeFree,
+    }
+
+    #[pallet::storage]
+    #[pallet::getter(fn is_free_proxy)]
+    pub type FreeProxyFlag<T: Config> = StorageValue<_, bool, ValueQuery>;
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        // in weight pallet_proxy::add_proxy(...) is accounted for
+        #[pallet::weight(< T as Config >::WeightInfo::add_free_proxy())]
+        pub fn add_free_proxy(
+            origin: OriginFor<T>,
+            delegate: T::AccountId,
+            proxy_type: T::ProxyType,
+            delay: T::BlockNumber,
+        ) -> DispatchResult {
+            let who = ensure_signed(origin)?;
+
+            let proxy_count = pallet_proxy::Proxies::<T>::get(&who).0.len();
+
+            ensure!(proxy_count == 0, Error::<T>::OnlyFirstProxyCanBeFree);
+
+            FreeProxyFlag::<T>::set(true);
+
+            let add_proxy_res = pallet_proxy::Pallet::<T>::add_proxy(
+                RawOrigin::Signed(who).into(),
+                delegate,
+                proxy_type,
+                delay,
+            );
```

Comment:
Can we use [add_proxy_delegate](https://github.com/paritytech/substrate/blob/1f0973906057ebdf58b88e3681847e79f6b9c7a9/frame/proxy/src/lib.rs#L640) directly here?

Reply:
why the goal is to have the same signature as pallet_proxy